tests/integration/run_tests.py

```diff
@@ -139,7 +145,7 @@ def run_tests(args, tests):
         n_to_run = 0
         selective = True
         for test_id, test in enumerate(tests.values()):
-            if len(test) == 3 and test[2] == "local":
+            if len(test) >= 3 and test[2] == "local":
```

Comment:
What's the significance of these magic constants, and what is the conditional here trying to do/enforce?

Reply:
This is to accommodate for the tuple increasing in size. The variable "local" is located in the 3rd element of the tuple and thus require the tuple to be at least of size 3. Regarding magic constants please see my comment above.

Reply:
Sure, but why does the tuple need to be increased.  What's the requirement driving this downstream?  and can there be some comments in here explaining the why so that reverse engineering of the whole test call stack isn't needed to figure it out.  At first glance from a new dev who's unfamiliar with this code it's just unclear why this conditional is here and also why the '>=' is the correct  condition.

Reply:
The increase in the tuple size is needed for a command to be ran after the completion of the unit test. This ending command is located at the 4th element of the tuple. It definitely possible to encode that information within the first element of the tuple, but it will require parsing and separating of the initial commands, and the ending commands later downstream.  This is needed specifically for `merlin server` command testing because it creates many different directories and files to stores config information. Without someway of cleaning up these configurations any subsequent `merlin server` commands will be using previous existing configurations. You are right that it would be great to have an additional comment would be great for future devs, so I have add one to explain the reason for the condition.

Reply:
Great, thanks for the explanation and the comment!